CacheManager.StackExchange.Redis.SelfCleaning.Tests/SelfCleaningRedisIntegrationTests.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using CacheManager.Core;
+using CacheManager.Core.Internal;
+using CacheManager.StackExchange.Redis.SelfCleaning.Core;
+using CacheManager.StackExchange.Redis.SelfCleaning.Timers;
+using Moq;
+using NUnit.Framework;
+using StackExchange.Redis;
+
+namespace CacheManager.StackExchange.Redis.SelfCleaning.Tests
+{
+    [TestFixture]
+    public class SelfCleaningRedisIntegrationTests
```

Comment:
1. Added. There isn't really a way to directly verify that the timer was disposed, so I used a workaround, see if it's understandable.
2.
2.1. I don't really see why. If no keys were found it would just return an empty enumerable and the method would pretty much end. Did you mean we should check it doesn't throw or something like that?
2.2. Added, although I feel I did it in a weird way since I couldn't think of any other way to verify that.

Reply:
1. ✌️  😎 
2. 
  2.1. I think so, I think we need to be sure that nothing unusual happens.
  2.2. I though about this flow:
a. setup to return null
b. put some key
c. wait (more than wait time)
d. verify that no cache item was expired (OnRemove wasnt raised).

dont you think that better?

both tests arent super important, but I think they should exist... No big deal, if you still dont agree with 2.1. and also want to keep 2.2. as you did it, thats fine

Reply:
Look at the tests I've added and see if that's what meant

Reply:
I've looked, you dont have test in which the server returns zero databases..

Actually, now as Im thinking about it - if the server returns zero databases it means that our configuration is bad, isnt it? Should we do something about it?

Reply:
> server returns databases
That doesn't happen in our code. Did you mean the `Servers` property in `RedisCacheHandle` returns no servers? If so, then that means there's no connection to Redis. I think the user should take care of that, otherwise they'll get an error either way. Did you refer to something else?

Reply:
No, I meant that Servers.Keys(_redisDatabase.Database) returns zero elements:

![image](https://user-images.githubusercontent.com/62211836/78533280-63002900-77f1-11ea-8bf0-21619151a7f5.png)